docs/examples/austin_issue/lab7/lab7.ipynb

```diff
@@ -0,0 +1,699 @@
+    "before we solved for $u$ and $h$ on the complete, non-staggered grid.\n",
+    "Figure [Decoupling](#lab7:fig:decoupling) shows the decoupling of the grids."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "Image(filename='images/decoupling.png',width='50%') "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div id='lab7:fig:decoupling'>\n",
+    "<b>Figure Decoupling</b>: The two staggered grids and the unstaggered grid. Note that the\n",
+    "unstaggered grid has two variables at each grid/time point whereas the\n",
+    "staggered grids only have one.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now consider the solution of the same problem on a staggered grid. The\n",
+    "set-up of the problem is slightly different this time; we are\n",
+    "considering 4 spatial points in our discussion instead of 5, shown in\n",
+    "Figure [Staggered Grid](#lab7:fig:stagger). We will also be using $h_{i}$ and $u_{i}$ to\n",
+    "denote the spatial points instead of $x = dx * i$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Image(filename='images/stagger.png',width='50%') "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lines_to_next_cell": 0
+   },
+   "source": [
+    "<div id='lab7:fig:stagger'>\n",
+    "<b>Figure Staggered Grid</b>: The staggered grid for the drop in the pond problem.\n",
+    "</div>\n",
+    "\n",
+    "The original equations, boundary and initial conditions are changed to\n",
+    "reflect the staggered case. The equations are changed to the following:\n",
+    "<div id='lab7:eq:staggerGrid1'>\n",
+    "(Staggered, Eqn 1)\n",
+    "$$\\frac {u_{i}(t+dt)-u_{i}(t-dt)}{2 dt} + g \\frac {h_{i + 1}(t) - h_{i}(t)}{dx} = 0$$\n",
+    "</div><div id='lab7:eq:staggerGrid2'>\n",
+    "(Staggered, Eqn 2)\n",
+    "$$\\frac {h_{i}(t+dt)-h_{i}(t-dt)}{2 dt} + H \\frac {u_{i}(t) - u_{i - 1}(t)}{dx} = 0$$\n",
+    "</div>\n",
+    "\n",
+    "The initial conditions are: At $t = 0$ and $t = dt$, all points have\n",
+    "zero elevation except at $h_{3}$, where $$h_{3}(0) = h_{0}$$\n",
+    "$$h_{3}(dt) = h_{3}(0) - h_{0} Hg \\frac{dt^2}{dx^2}$$ At $t = 0$ and\n",
+    "$t = dt$, all points have zero velocity except at $u_{2}$ and $u_{3}$,\n",
+    "where $$u_{2}(dt) = - h_{0} g \\frac{dt}{dx}$$\n",
+    "$$u_{3}(dt) = - u_{2}(dt)$$ This time we assume there is a wall at\n",
+    "$u_{1}$ and $u_{4}$, so we will ignore the value of $h_{1}$. The\n",
+    "boundary conditions are: $$u_{1}(t) = 0$$ $$u_{4}(t) = 0$$\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lines_to_next_cell": 0
+   },
+   "source": [
+    "### Problem One\n",
+    "[lab7:prob:staggered]:(#Problem-One) \n",
```

Comment:
@phaustin thanks for the feedback

> I'm moving one of my courses from nbsphinx to myst-nb, and have hit an issue with a commented link causing myst to hang. Specifically, this line is ignored by nbsphinx, but causes the myst parser to go into an infinite loop printing the message:

> "WARNING: /Users/phil/repos/MyST-NB/docs/examples/austin_issue/lab7/lab7.ipynb:2:1 {'cell_index': 28} ignoring duplicate link definition 'lab7:prob:staggered'"

Ok, firstly @choldgraf note how the warning gives the correct notebook cell and line number 😄 (this is currently only for the source - Markdown AST phase, not the Markdown -> docutils AST phase).

Secondly, going by the warning message, the error actually lies here, not to where you linked to. Checking it with mistletoe (the markdown parser), it does replicate this error:

```console
$ mistletoe
mistletoe [version 0.10.0] (interactive)
Type Ctrl-D to complete input, or Ctrl-C to exit.
>>> [lab7:prob:staggered]:(#Problem-One) 
... > Modify
:2:1 ignoring duplicate link definition 'lab7:prob:staggered'
...
```

The issue lies in the transition from link definition to quote block , somewhere around here: https://github.com/ExecutableBookProject/mistletoe-ebp/blob/d7247b52df3ff50dc71aef9ec3f8e7b316b03813/mistletoe/block_tokens.py#L821

This is **only** caused if there is no new line between the link definition and the quote, 
whereby the line number for some reason is not being progressed (so it reads the same line infinitely)

As a stop gap, if you add a blank line, there should be no problem:

```
[lab7:prob:staggered]:(#Problem-One)

> ...
```

Reply:
Yep see  ExecutableBookProject/mistletoe-ebp#13 

Literally if you just change `[lab7:prob:staggered]:(#Problem-One) \n` to `[lab7:prob:staggered]:(#Problem-One)\n`, then it is ok!?